File: 0112-path-sum/0112-path-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean res;
    
    public boolean hasPathSum(TreeNode root, int targetSum) {
        res = false;
        if(root==null){
            return res;
        }
        int sum = 0;
        pathSum(root,sum,targetSum);
        return res;
    }

    public void pathSum(TreeNode root, int sum, int targetSum){
        if(root==null){
            return;
        }
        sum+=root.val;
        if(sum==targetSum && root.left==null && root.right==null){
            res = true;
        }
        pathSum(root.left,sum,targetSum);
        pathSum(root.right,sum,targetSum);
    }
}

File: 0112-path-sum/0112-path-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    bool fin = false;

    bool hasPathSum(TreeNode* root, int targetSum) {
        traverse(root,0,targetSum);
        return fin;
    }

    void traverse(TreeNode* root, int sum,int targetSum){
        if(root == nullptr){
            return;
        }
        sum += root->val;
        if((root->right == nullptr) && (root->left == nullptr)){
            if(sum == targetSum){
                fin = true;
            }
        }
        traverse(root->left,sum,targetSum);
        traverse(root->right,sum,targetSum);
    }
};